archive/test_archive.py

```diff
@@ -0,0 +1,243 @@
+""" Unit tests to test archive functions. """
+import os
+import unittest
+from unittest.mock import MagicMock, patch
+
+from psycopg2.extensions import connection
+from psycopg2.extras import RealDictCursor
+
+
+from db_connection import get_connection, get_cursor, execute
+from clean_real_time_trains import (
+    get_all_station_ids,
+    get_month_old_waypoints,
+    compute_avg_delay_for_station,
+    compute_cancellation_count_for_station,
+    insert_performance_archive,
+    delete_cancellation,
+    delete_waypoint,
+    get_table_size
+)
+
+
+class ArchiveTests(unittest.TestCase):
+    """ Class for testing functions from the archive directory. """
+
+    @patch.dict(
+        os.environ,
+        {
+            "DB_IP": "localhost",
+            "DB_NAME": "test_db",
+            "DB_USERNAME": "test_username",
+            "DB_PASSWORD": "test_password",
+            "DB_PORT": "test_port",
+        },
+    )
+    @patch("db_connection.connect")
+    @patch("db_connection.load_dotenv")
+    def test_get_connection(self, mock_load_dotenv, mock_connect: connection):
+        """ Tests get_connection returns connection. """
+
+        mock_connect.return_value = "mock_connection"
+        db_connection = get_connection()
+
+        assert db_connection == "mock_connection"
+        mock_load_dotenv.assert_called_once()
+        mock_connect.assert_called_once()
+
+    @patch("db_connection.get_connection")
+    def test_get_cursor(self, mock_get_connection: connection):
+        """ Tests get_cursor returns cursor. """
+
+        mock_connection = MagicMock()
+        mock_get_connection.return_value = mock_connection
+        db_cursor = get_cursor(mock_connection)
+
+        mock_connection.cursor.assert_called_once_with(
+            cursor_factory=RealDictCursor)
```

Comment:
Why are you still using realdictcursor here?

Reply:
I DONT KNOW OKAY

Reply:
so I can fetch data as a dict instead of a tuple

Reply:
Fine by me